content/klc/S2.1.adoc

```diff
@@ -2,15 +2,15 @@
 brief = "General symbol naming guidelines"
 +++
 
-. Library name should not be duplicated in symbol name
+. Library naming should not be duplicated in footprint name
 . If symbol with same name exists for multiple manufacturers, the manufacturer name is written first
-. Specific manufacturer name (for atomic parts)
-. Type of symbol (for generic parts)
+. Fully specified symbols are named based on the manufacturer part number
+.. If and only if multiple manufacturers build compatible devices then the common base part number is used with clear name suffix for the footprint (e.g. `L78L05_TO92`)
```

Comment:
If the MPNs are identical then there's no problem. This only applies if the MPN suffix for a compatible part with the same MPN and package have conflicts, like 'D' and 'M' suffixes both being for compatible SOIC packages by some IC vendors. So this can be clarified that only when the MPN, including the suffix to define the package, is in conflict does this system need to be employed.

Reply:
Do you really want to be responsible to check every possible manufacturer? With the rule as stated we can simply say "this is produced by many manufacturers who in theory could use different numbering schemes so please use cleartext"

Reply:
What I am trying to say is that if the MPN is identical between vendors then this suffix of package isn't needed. We could go with `L78L05Z` and not `L78L05_TO92`, for example, if all vendors use the suffix 'Z' for the TO92 package. But if the functionally- and package-equivalent part is sold as `L78L05A` and `L78L05B` then we use this scheme.

Only if we are alerted to this issue then we should go with this scheme but it doesn't need to be expected. I don't intend to intimate that librarians must scour the internet for every possible PR. We have seen this issue come up just in a few places like 'classic' regulators and opamps/comparators, so we probably know when it's a possibility. I only mean to try and limit the use of this rule as much as possible.

Reply:
Generally, I'm with @poeschlr on this, compatible parts that even share package suffixes among manufactures are probably a rare find so clear text is the safe option.

But is an 78xx really the best example here? It is probably one of the most well known parts in the world but has five dozen manufacturers with ten dozen different part numbers each (different prefixes, different infixes vor low dropout variants, different suffixes for output tolerance, different packages and whatnot)

I can't think of a good one right know, but perhaps at least another (counter-) example should be added.

Reply:
Here we have i think a different philosophy. For now i will go with your wording as i guess this is how it is done right now (You are after all more active on the symbol side). But i will for sure open an issue about it for discussing if this is the best approach. (Listing pros and cons of both approaches such that we can then select which one to use.)

Reply:
@chschlue
https://github.com/KiCad/kicad-symbols/pull/1081 is one example. `DG419` is another where ADI uses the suffix `N` and Maxim uses `J` (our library uses a `_Maxim` suffix which isn't described here at all so it may be deprecated). Or `TL072` where ST uses `D`, TI uses `DR`, and Diodes Inc uses `SG`. It's really those common ICs where I've seen it. (Note that when scanning through the library I think this might be mis-used and we don't really need as many examples, or what is there is to maintain compatibility with manufacturers that don't actively make the parts anymore, like `MC1408`.)

@poeschlr 
I would just like to replace `build compatible devices` with `build compatible devices with differing suffixes` to reduce this rule's use as much as possible.

Reply:
Is already written this way at the point of this answer.

Reply:
I feel so annoying about continuing with this topic, but I still have a couple of comments.
1. The word `compatible` isn't quite right to me since it gives the impression of "similar, but not necessarily identical". We want to invoke this rule only when the "base" MPN is identical, right?
1. Also, the phrase about describing the suffix seems overwrought to me.

May I suggest `If multiple manufacturers produce equivalent parts where the part number is the same except for the suffix that specifies the package, then the common base part number is kept and suffixed with the footprint name (e.g. `L78L05_TO92`)`?

https://github.com/KiCad/kicad-symbols/pull/2076 is a recent example using parts from datasheets at http://www.ti.com/lit/ds/symlink/sn74lvc1g66.pdf and https://assets.nexperia.com/documents/data-sheet/74LVC1G66.pdf.

Reply:
That proposal looks better to me, too.

Reply:
@evanshultz 
I objected to the `L78L05_TO92` example before but your suggested wording transforms it from questionable to flat out wrong.

Reply:
I think there are better examples too, but I honestly didn't look into the section into the parenthesis very much. I was trying to make the sentence read better and be specific.

Could you please enlighten us about my wrongness? Maybe with a suggestion of your own? Perhaps you object to the use of `footprint name` before the open paren and would prefer `simplified footprint name`? Re-reading it that sounds better to me.

Reply:
I'm not opposed to the suggested sentence. The problem is: The example contradicts it because equivalent parts all differ at least in prefix, too. Correct me if I'm wrong, but AFAIK an L78* is invariably manufactured by ST. Equivalent parts use MC78*, µA78*, ...
You gave better examples yourself, TL072, DG419, ...

I get that the "base" MPN of the example is 78[Vout], but it's at least misleading when the rule explicitly says package _suffix_ only.

Reply:
Meaning my original wording fits the current lib better as the `L78L05_TO92` is in our lib.

Reply:
Chicken and egg... all the other examples are already in the lib, sporting wrong names.

Reply:
Hmm... That's not how I interpret the conversation, Rene.

I've suggested different wording. Matthis concurred. Christian is OK with the wording too, but thinks the example given is a poor one. I agree, but from a different perspective.

Would changing the sentence as shown above and _also_ changing the example above satisfy everyone? What objection is there to that idea? I originally noted this paradigm at the link below for `MC33078`. `DG419` and `TL072` are also noted above.

Reply:
LF398 would work in it's current state. Caveat: TI seems to be only manufacturer left.

Reply:
Oh, I wasn't aware this might be unclear. Short version:
Yes, I like Evan's wording better.
Yes, I think the current example is bad.
No, I don't have an optimal example that doesn't conflict with the current lib.

Reply:
The problem is that the change in KLC was intended to capture the current state of the lib. Changing the example would not change the fact that this symbol is in the library. If it contradicts the rule as an example then it also contradicts it just by existing in the lib. So we have two options. Word the rule such that we have the wiggle room such that this symbol is correct or strengthen the wording as suggested and make this symbol non compliant.

Reply:
You are right in that this is definitely a color-of-the-bikeshed discussion. I certainly don't mean to block the fully specified vs. generic stuff over a `fucking example`.

But a more general question: Is KLC really supposed to capture the current state of the library, which is a little rough around the edges to say the least?
Shouldn't the library convention be the set point rather than the process value, if you'll excuse the control loop analogy?

Reply:
You are right the KLC should point to a future goal.

Here it is however meant to clarify an already existing rule. That was in part created exactly for covering the part listed in the example.

The pull request that added this type of suffix to it was one of the first that i reviewed after evan and jan made this suggestion in the first place. So to me the fact that it now violates the rule shows that the current wording does not cover the original intent and would therefore require more than a change in the minor release number.

Also with the stronger wording is there even a significant number of symbols left where this would be allowed? Because maybe a better future goal (for v6) would be to get rid of this exception and require fully specified with aliases for differing manufacturer conventions. (This would be even easier for us than my wording with wiggle room as we then don't even need to check if multiple manufacturers make this part.)

Reply:
Now I've got it. Probably. Hopefully.
Keep the wording and the fucking example. Nothing to see here. Please disperse.

Is there a clear cut compilation of what is set in stone / planned / wishful thinking for v6 anywhere?
I wasn't able to find much, but maybe I'm just too dumb for Launchpad.

Reply:
In this case it has nothing to do with what the devs do but it is simply a matter of fact that massive renaming of symbols (which the changed wording seems to require) can simply not happen during a minor release. So it would really need to wait till v6.

---

Really i don't get why this discussion is so heated here. You guys wanted me to make the wording stronger, then you discovered that the new wording (suggested by you) contradicts the example that has always been in this rule and now you are mad at me for pointing this out.

Reply:
@chschlue
I may be too dumb for Launchpad as well. I also often can't find what I want using the website to search the dev mailing list archive.